### ISSUE TITLE
Feature: Output debug and warning messages

### ISSUE THREAD

**Issue by glenn2223** [glenn2223/vscode-live-sass-compiler#89]
**Nothing here is set in stone. Open for discussion, but happy with the current plans**

**Please keep comments related to this issue. Want something else adding/sorting? Please open a new issue**

---

### Proposal details

**Issue**

There is currently not output when the compiler encounters any `@warn` or `@debug` messages.

**Resolution**
The team at dart-sass are currently working on adding a logger feature to the Javascript API. This will expose a means of outputting these types of messages.

The below items are being monitored and, once available, I will implement a change
- [x] sass/sass#3074 - PR - **MERGED**
- [x] [sass/sass#2979](https://github.com/sass/sass/issues/2979#issuecomment-858177193) - Issue - **ACCEPTED** _(starting work)_
- [x] sass/sass#3056 - Issue - **ACCEPTED** _(above is waiting on this to bundle in)_
---

### To do list

- [x] WAIT & monitor the above
- [x] Apply code changes
- [x] ~Doc changes~ (not required)

**Comment by glenn2223** [comment]
Just have a couple of improvements to make to the logger. I will release an alpha now though as the functionality is there

**Comment by glenn2223** [comment]
So v5.2.0-alpha.1 is available for testing. Download it below or from the `ahead`  branch

[live-sass-5.2.0-alpha.1.zip](https://github.com/glenn2223/vscode-live-sass-compiler/files/7800163/live-sass-5.2.0-alpha.1.zip)

----

Note, improvements are still outstanding as of yet

**Comment by lucianahanan** [comment]
Hello, good evening. I wonder, have you added the @debug and @warn outputs or not there yet? Today I'm trying to print a simple variable here, but yet nothing shows in vscode output tab or in browser. Am I forgetting to do something, maybe?

**Comment by glenn2223** [comment]
Hey @lucianahanan. Yes, it is in there!

![Warning message in code which is reflected in the output window](https://user-images.githubusercontent.com/33450392/187644539-e15f2def-9a64-4a2e-8184-0c7d3514c167.png)

They output the message if the 'showOutputWindowOn` is set to it's specific level (or higher). So `@debug` messages will show when it's set to `Debug` or higher and `@warn` when it's set to `Warning` or higher (so `Warning` or `Error`)

However, I'd suggest using `@warn` as debug messages may get lost in the output

**Comment by lucianahanan** [comment]
I tried to enable all kinds of messages, and then tried to use @debug and @warn, but still, none of them shows anywhere! Can you please show me how exactly you did the settings in order to have it working?

**Comment by glenn2223** [comment]
I have the below screenshots showing the setup of a test workspace - which I have zipped up ([Issue 89.zip](https://github.com/glenn2223/vscode-live-sass-compiler/files/9492135/Issue.89.zip)).

![image](https://user-images.githubusercontent.com/33450392/188515818-80fa0e1d-fed6-4f5f-b714-ee03121fe623.png)

![image](https://user-images.githubusercontent.com/33450392/188515883-d523814d-4132-4e90-acc5-997d7011b8bb.png)

Your setting needs to be as seen, then when you click `Watch SASS` or save a file while the extension is watching it will show the output window with your `@debug` or `@warn` message.

Does this make sense?
